File: app/src/main/java/com/takusemba/multisnaprecyclerviewsample/MainActivity.java
package com.takusemba.multisnaprecyclerviewsample;

import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.HorizontalScrollView;

public class MainActivity extends AppCompatActivity {

    String[] titles = {
            "HorizontalLayout",
            "VerticalLayout",
    };

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        findViewById(R.id.horizontal).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this, HorizontalActivity.class));
            }
        });

        findViewById(R.id.vertical).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivity(new Intent(MainActivity.this, VerticalActivity.class));
            }
        });
    }
}

File: app/src/main/java/com/takusemba/multisnaprecyclerviewsample/MainActivity.kt
package com.takusemba.multisnaprecyclerviewsample

import android.content.Intent
import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)

    findViewById<TextView>(R.id.horizontal).setOnClickListener {
      startActivity(Intent(this@MainActivity, HorizontalActivity::class.java))
    }

    findViewById<TextView>(R.id.vertical).setOnClickListener {
      startActivity(Intent(this@MainActivity, VerticalActivity::class.java))
    }
  }
}
